symengine/monomials.cpp

```diff
@@ -8,6 +8,7 @@ namespace SymEngine {
 void monomial_mul(const vec_int &A, const vec_int &B, vec_int &C)
 {
     size_t n = A.size();
+    C.resize(n);
```

Comment:
Just run the benchmarks to see.

You can add a check in Debug mode using SYMENGINE_ASSERT, that's a good idea.

However, you should not be resizing it here. As I said, try to write a test first, that segfaults. Then we'll figure out how to fix it.